XamarinCommunityToolkit/Converters/StringToDoubleConverter.shared.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	///     Converts The Provided string into the equivalent Double.
+	/// </summary>
+	public class StringToDoubleConverter : ValueConverterExtension, IValueConverter
+	{
+		/// <summary>
+		///     Converts The Provided string into the equivalent double.
+		/// </summary>
+		/// <param name="value">The value to convert.</param>
+		/// <param name="targetType">The type of the binding target property.</param>
+		/// <param name="parameter">Additional parameter for the converter to handle. Not implemented.</param>
+		/// <param name="culture">The culture to use in the converter.</param>
+		/// <returns>The double value of the provided string</returns>
+		public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+		{
+			if (value is string numstr && !string.IsNullOrWhiteSpace(numstr))
+				return double.TryParse(numstr, out var number) ? number : throw new ArgumentException("Value is not a valid double", nameof(value));
+			else
+				throw new ArgumentException("Value is not a valid integer", nameof(value));
```

Comment:
Why not just
```
			=> value is string numstr && double.TryParse(numstr, out var number)
				? number
				: throw new ArgumentException("Value is not a valid double", nameof(value));
```

Reply:
I guess here it is useless but i always try to make sure that my strings are not empty before using them.
Thank you for pointing that out i will fix them